config/psptoolchain-allegrex-config.sh

```diff
@@ -3,7 +3,7 @@
 PSPTOOLCHAIN_ALLEGREX_BINUTILS_REPO_URL="https://github.com/pspdev/binutils-gdb.git"
 PSPTOOLCHAIN_ALLEGREX_BINUTILS_DEFAULT_REPO_REF="allegrex-v2.44"
 PSPTOOLCHAIN_ALLEGREX_GCC_REPO_URL="https://github.com/pspdev/gcc.git"
-PSPTOOLCHAIN_ALLEGREX_GCC_DEFAULT_REPO_REF="allegrex-v14.2.0"
+PSPTOOLCHAIN_ALLEGREX_GCC_DEFAULT_REPO_REF="allegrex-v15.1.0"
```

Comment:
We can also use the 15.1.1 branch I rebased the changes from the 15.1.0 branch on, which works great as well:
```suggestion
PSPTOOLCHAIN_ALLEGREX_GCC_DEFAULT_REPO_REF="allegrex-v15.1.1"
```
Either way this is ready to be merged, I don't see any issues so far. OceanPop worked, the psp-packages repo builds.